crates/rome_cli/src/commands/format.rs

```diff
@@ -0,0 +1,468 @@
+            Ok(entry) => entry,
+            Err(err) => {
+                ctx.push_diagnostic(err);
+                continue;
+            }
+        };
+
+        let path = entry.path();
+        let file_id = ctx.acquire_file_id(path.clone());
+
+        let file_type = match entry.file_type().with_file_id(file_id) {
+            Ok(file_type) => file_type,
+            Err(err) => {
+                ctx.push_diagnostic(err);
+                continue;
+            }
+        };
+
+        if file_type.is_dir() {
+            let ctx = ctx.clone();
+            scope.spawn(move |scope| {
+                handle_dir(scope, ctx, &path, file_id);
+            });
+            continue;
+        }
+
+        if file_type.is_file() {
+            // Performing this check here lets us skip scheduling unsupported
+            // files entirely as well as silently ignore unsupported files when
+            // doing a directory traversal but printing an error message if the
+            // user explicitly requests an unsupported file to be formatted
+            if !ctx.app.can_format(&RomePath::new(&path)) {
+                continue;
+            }
+
+            let ctx = ctx.clone();
+            scope.spawn(move |_| {
+                handle_file(ctx, &path, file_id);
+            });
+            continue;
+        }
+
+        ctx.push_diagnostic(Diagnostic::error(file_id, "IO", "unhandled file type"));
+    }
+}
+
+/// This function wraps the [format_file] function implementing the formatting
+/// in a [catch_unwind] block and emit diagnostics in case of error (either the
+/// formatting function returns Err or panics)
+fn handle_file(ctx: TraversalContext, path: &Path, file_id: FileId) {
+    let params = FormatFileParams {
+        app: ctx.app,
+        options: ctx.options,
+        is_check: ctx.is_check,
+        path,
+        file_id,
+    };
+
+    match catch_unwind(move || format_file(params)) {
+        Ok(Ok(errors)) => {
+            if errors.is_empty() {
+                ctx.add_formatted();
+            } else {
+                for error in errors {
+                    ctx.push_diagnostic(error);
+                }
+            }
+        }
+        Ok(Err(err)) => {
+            ctx.push_diagnostic(err);
+        }
+        Err(err) => {
+            let msg = match err.downcast::<String>() {
+                Ok(msg) => format!("formatting panicked: {msg}"),
+                Err(_) => String::from("formatting panicked"),
+            };
+
+            ctx.push_diagnostic(Diagnostic::error(file_id, "Panic", msg));
+        }
+    }
+}
+
+struct FormatFileParams<'a> {
+    app: &'a App,
+    options: FormatOptions,
+    is_check: bool,
+    path: &'a Path,
+    file_id: FileId,
+}
+
+/// This function performs the actual formatting: it reads the file from disk
+/// (or mmap it), parse and format it then either write the result back or
+/// compare it to the original content and emit a diagnostic in check mode
```

Comment:
What does `mmap` mean? It's not standard english, so we should add a link to it (if it's an acronym) or reword to make it simpler to understand.
```suggestion
/// This function performs the actual formatting: it reads the file from disk
/// (or mmap it), parse and format it; then, it either writes the result back or
/// compares it with the original content and emit a diagnostic
```

I removed "in check mode" because if we weren't in check mode, it would imply there was another scenario, which is not. It's either check mode (comparison) or not (write to file)